packages/vite/src/node/plugins/css.ts

```diff
@@ -562,20 +568,48 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           const cssFileName = ensureFileExt(cssAssetName, '.css')
 
           chunkCSS = resolveAssetUrlsInCss(chunkCSS, cssAssetName)
-          chunkCSS = await finalizeCss(chunkCSS, true, config)
-
-          // emit corresponding css file
-          const referenceId = this.emitFile({
-            name: path.basename(cssFileName),
-            type: 'asset',
-            source: chunkCSS,
+          const emitName = path.basename(cssFileName)
+
+          emitTasks.add({
+            name: emitName,
+            emit: async () => {
+              chunkCSS = await finalizeCss(chunkCSS, true, config)
+
+              // emit corresponding css file
+              const referenceId = this.emitFile({
+                name: emitName,
+                type: 'asset',
+                source: chunkCSS,
+              })
+              const originalName = isPreProcessor(lang)
+                ? cssAssetName
+                : cssFileName
+              const isEntry = chunk.isEntry && isPureCssChunk
+              generatedAssets
+                .get(config)!
+                .set(referenceId, { originalName, isEntry })
+              chunk.viteMetadata!.importedCss.add(this.getFileName(referenceId))
+            },
           })
-          const originalName = isPreProcessor(lang) ? cssAssetName : cssFileName
-          const isEntry = chunk.isEntry && isPureCssChunk
-          generatedAssets
-            .get(config)!
-            .set(referenceId, { originalName, isEntry })
-          chunk.viteMetadata!.importedCss.add(this.getFileName(referenceId))
+
+          // wait for collecting all emitFile
+          await sortedEmitTasks
```

Comment:
`sortedEmitTasks` is always a simple array. IIUC this `await` never waits.

Reply:
Emm, it will await even though `sortedEmitTasks` is an array.